src/Maestro/maestro-angular/src/app/widget/side-bar-channel/side-bar-channel.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, Input, OnInit, OnDestroy } from "@angular/core";
+import { NavigationEnd, Router, ActivatedRoute } from "@angular/router";
+import { Observable, SubscriptionLike, of } from "rxjs";
+import { filter, map, switchMap, concat } from "rxjs/operators";
+
+import { prettyRepository } from "../../util/names";
+import { MaestroService } from 'src/maestro-client';
+import { Channel, DefaultChannel } from 'src/maestro-client/models';
+
+@Component({
+  selector: "mc-side-bar-channel",
+  templateUrl: "./side-bar-channel.component.html",
+  styleUrls: ["./side-bar-channel.component.scss"],
+})
+export class SideBarChannelComponent implements OnInit, OnDestroy {
+
+  public constructor(private maestro: MaestroService, private router: Router) { }
+  @Input() public channel!: Channel;
+
+  public isCollapsed = true;
+
+  public branches$!: Observable<DefaultChannel[]>;
+
+  public trimName = prettyRepository;
+
+  private routeSubscription?: SubscriptionLike;
+
+  public ngOnInit() {
+    this.branches$ = this.maestro.defaultChannels.listAsync(undefined, this.channel.id);
+
+    // Get the current route state, and concat it with any changes
+    this.routeSubscription =
+      of(this.router.routerState.root).pipe(
```

Comment:
>of(this.router.routerState.root).pipe( [](start = 6, length = 38)

But... of(foo).pipe(stuff).subscript(bar) is just... bar(stuff(foo)).

Why make an observable of a fixed value with of?  My understanding of rxjs is that it doesn't do anything.

Reply:
This is to take the current value, then concat it with the new value when a NavigationEnd happens. this of(stuff).pipe(concat(stuff)) == emit the value of `stuff` once now, then again whenever navigation end happens.

I think the of().pipe syntax is nicer than concat(<multiline thing>)(of(stuff))